File: frontend/src/app/core/infrastructure/repositories/activity-log.repository.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ActivityLog } from '../../domain/models/activity-log.model';

export interface PageResponse<T> {
    content: T[];
    totalElements: number;
    totalPages: number;
    size: number;
    number: number;
}

@Injectable({
    providedIn: 'root'
})
export class ActivityLogRepository {
    private readonly baseUrl = '/api/activity-logs';

    constructor(private http: HttpClient) { }

    getUserLogs(userId: number, page: number = 0, size: number = 20): Observable<PageResponse<ActivityLog>> {
        const params = new HttpParams()
            .set('page', page.toString())
            .set('size', size.toString());

        return this.http.get<PageResponse<ActivityLog>>(`${this.baseUrl}/user/${userId}`, { params });
    }

    getLogsByAction(action: string, page: number = 0, size: number = 20): Observable<PageResponse<ActivityLog>> {
        const params = new HttpParams()
            .set('page', page.toString())
            .set('size', size.toString());

        return this.http.get<PageResponse<ActivityLog>>(`${this.baseUrl}/action/${action}`, { params });
    }

    getLogsByTarget(target: string, page: number = 0, size: number = 20): Observable<PageResponse<ActivityLog>> {
        const params = new HttpParams()
            .set('page', page.toString())
            .set('size', size.toString());

        return this.http.get<PageResponse<ActivityLog>>(`${this.baseUrl}/target/${target}`, { params });
    }

    getLogsByDateRange(startDate: Date, endDate: Date, page: number = 0, size: number = 20): Observable<PageResponse<ActivityLog>> {
        const params = new HttpParams()
            .set('startDate', startDate.toISOString())
            .set('endDate', endDate.toISOString())
            .set('page', page.toString())
            .set('size', size.toString());

        return this.http.get<PageResponse<ActivityLog>>(`${this.baseUrl}/date-range`, { params });
    }
}

File: backend/src/main/kotlin/br/com/vrsoftware/vragentpdv/repository/ActivityLogRepository.kt
package br.com.vrsoftware.vragentpdv.repository

import br.com.vrsoftware.vragentpdv.model.ActivityLog
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import java.time.Instant

interface ActivityLogRepository {
    fun save(activityLog: ActivityLog): ActivityLog
    fun findByUserId(userId: Long, pageable: Pageable): Page<ActivityLog>
    fun findByAction(action: String, pageable: Pageable): Page<ActivityLog>
    fun findByUserIdAndAction(userId: Long, action: String, pageable: Pageable): Page<ActivityLog>
    fun findByTarget(target: String, pageable: Pageable): Page<ActivityLog>
    fun findByDateRange(startDate: Instant, endDate: Instant, pageable: Pageable): Page<ActivityLog>
}